qualtran/bloqs/hamiltonian_simulation/hamiltonian_simulation_by_gqsp_test.py

```diff
@@ -75,7 +76,8 @@ def verify_hamiltonian_simulation_by_gqsp(
     N = H.shape[0]
 
     W_e_iHt = HamiltonianSimulationByGQSP(W, t=t, precision=precision)
-    result_unitary = cirq.unitary(W_e_iHt)
+    # This cirq.unitary call is 4-5x faster than tensor_contract.
+    result_unitary = cirq.unitary(BloqAsCirqGate(W_e_iHt))
```

Comment:
can you open an issue to track the discrepancy. Also: what's the magnitude of the difference? hundreds of milliseconds or seconds? 

presumably since we're throwing away most of this matrix; a tensor contraction could theoretically be more performant for this use case

Reply:
#1336